contrib/format-xml/src/test/java/org/apache/drill/exec/store/xml/xsd/TestXSDSchema.java

```diff
@@ -61,38 +62,41 @@ public void testComplexXSD() throws Exception {
     File complex_xsd = DrillFileUtils.getResourceAsFile("/xsd/complex.xsd");
     TupleMetadata schema = DrillXSDSchemaUtils.getSchema(complex_xsd.getPath());
 
-    TupleMetadata expectedSchema = new SchemaBuilder()
-        .addNullable("comment", MinorType.VARCHAR)
-        .addMap("purchaseOrder")
+    SchemaBuilder sb1 = new SchemaBuilder();
+    MapBuilder sb2 = sb1
+        .addNullable("comment", MinorType.VARCHAR) // global comment element
+        .addMap("purchaseOrder") // global purchaseOrder element
           .addMap("shipTo")
             .addNullable("name", MinorType.VARCHAR)
```

Comment:
I have a question about these child map definitions. 

Can these be shared? So, since both shipTo and billTo have the same sub-structure, could we share one collection of the sub-fields metadata?

It seems these MapBuilder objects have a parent pointer, which makes sharing impossible.

Reply:
Good question... I'm not really sure.